### ISSUE TITLE
Implement TPM2_Commit

### ISSUE THREAD

**Issue by akakou** [google/go-tpm#287]
It needs when implementing ECDAA(e.g. https://github.com/google/go-tpm/issues/278#issuecomment-1148113966).

>[tpm2_commit](https://manpages.ubuntu.com/manpages/jammy/man1/tpm2_commit.1.html)(1)  -  Performs the first part of an ECC anonymous signing operation.  The TPM will perform the point multiplications on the  provided  points  and  return  intermediate signing  values.  The signing key is an ECC key.  The key cannot be a sign+decrypt key and must have an anonymous signing scheme.  TPM_ALG_ECDAA  is  the  only  supported  anonymous scheme.

https://manpages.ubuntu.com/manpages/jammy/man1/tpm2_commit.1.html

**Comment by akakou** [comment]
Now I'm implementing the test code of TPM2_Commit, but not work.
Do you have any idea to solve this error?

```
go test -run TestCommit
--- FAIL: TestCommit (0.01s)
    commit_test.go:102: could not commit: TPM_RC_AUTH_MISSING: command requires an authorization session for handle and it is not present.
FAIL
exit status 1
FAIL    github.com/google/go-tpm/direct/tpm2    0.012s
```

https://github.com/akakou/go-tpm/blob/tpm2-commit/direct/tpm2/commit_test.go

**Comment by akakou** [comment]
Thank you. It's one of the bugs!
But it has never worked still...and panic with the error following:

```
--- FAIL: TestCommit (0.00s)
    commit_test.go:102: could not commit: TPM_RC_SIZE (handle 0): structure is the wrong size
FAIL
exit status 1
FAIL    github.com/google/go-tpm/direct/tpm2    0.008s
```

Do you have any idea?

**Comment by akakou** [comment]
Thank you! You are right and I fixed them.

Specifically, I fixed them as:
1. Set `x=0x1`, `y = 0x2` to P1.
    - It's the generator of ECP.
    - 5.2.7 BN P256 (https://trustedcomputinggroup.org/wp-content/uploads/TCG_Algorithm_Registry_Rev_1.22.pdf)
2. Use annotation to express the size of TPM2B_ECC_POINT

**Comment by akakou** [comment]
But the program calls the same error...

**Comment by akakou** [comment]
I compared worked TPM2_Commit(tpm2-tools) and this library.
Mainly, I compared what binaries each library sent to TPM.


These are the binaries:

go-tpm
> \x80\x02\x00\x00\x00\x40\x00\x00\x01\x91\x40\x00\x00\x01\x00\x00\x00\x09\x40\x00\x00\x09\x00\x00\x00\x00\x00\x00\x07\x00\x03\x00\x01\x03\x00\x00\x00\x1a\x00\x23\x00\x04\x00\x04\x00\x72\x00\x00\x00\x10\x00\x1a\x00\x0b\x00\x00\x00\x10\x00\x10\x00\x00\x00\x00

tpm2-tools
> \x80\x02\x00\x00\x00\x65\x00\x00\x01\x8b\x80\xff\xff\xff\x00\x00\x00\x49\x02\x00\x00\x00\x00\x20\x14\xe2\x02\x60\x66\x57\x5f\x9c\xab\x50\x13\x21\x36\x37\xc3\xfd\x24\x86\xf8\xae\x9f\xa1\x0b\x62\x3e\x4a\x00\xa8\x91\x8d\xf3\x10\x01\x00\x20\x2f\xc0\x53\x68\x30\xa0\x0b\xc3\x88\x5f\x43\xed\x73\x00\xa8\x79\x87\xae\x96\x5a\x78\xa3\xe1\xda\x40\xa4\xa0\x0d\xae\x11\x4f\xca\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00

**Comment by akakou** [comment]
Sorry...I had added the unnecessary parameter to the request of TPM2_Commit.
Specifically, we should not add the `count` to the request.

**Comment by akakou** [comment]
@chrisfenner 

So I fixed it and it work.
Could you check the PR (#296)  and merge it?

**Comment by akakou** [comment]
https://github.com/google/go-tpm/pull/296#discussion_r930604458

**Comment by akakou** [comment]
It is implemented at  #296, so I closed this issue.
